File: src/main/scala/org/tindalos/principle/domain/expectations/cumulativedependency/DoubleThresholder.scala
package org.tindalos.principle.domain.expectations.cumulativedependency

abstract class DoubleThresholder(var threshold: Double = 0)
case class ACD(_threshold: Double = 0) extends DoubleThresholder(_threshold)
case class NCCD(_threshold: Double = 0) extends DoubleThresholder(_threshold)
case class RACD(_threshold: Double = 0) extends DoubleThresholder(_threshold)

File: src/main/java/org/tindalos/principle/domain/expectations/DoubleThresholder.java
package org.tindalos.principle.domain.expectations;

/**
 * Interface for thresholders that use a double value as threshold.
 * Represents expectations with floating-point threshold values.
 */
public interface DoubleThresholder {

    /**
     * Gets the threshold value.
     *
     * @return the threshold value as a double
     */
    double threshold();
}
